src/sdl2/video.rs

```diff
@@ -1177,23 +1177,13 @@ impl Window {
         }
     }
 
-    pub fn surface<'a>(&'a self, _e: &'a EventPump) -> Result<&'a SurfaceRef, String> {
+    pub fn surface<'a>(&'a mut self, _e: &'a EventPump) -> Result<Surface<'a>, String> {
```

Comment:
Wait, it used to return a &SurfaceRef but now you return a Surface ? [You cannot free the surface that you've been given here](https://wiki.libsdl.org/SDL_GetWindowSurface#Remarks). I think that is the reason behind SurfaceRef and Surface in the end, I don't think there is an actual solution to make this work without creating another type.